tmt/steps/execute/__init__.py

```diff
@@ -450,21 +450,24 @@ def load_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Resul
         # Check the test result
         self.debug(f"tmt-report-results file '{report_result_path} detected.")
 
+        hierarchy = {'skip': 1, 'pass': 2, 'warn': 3, 'fail': 4}
         with open(report_result_path) as result_file:
-            result_list = [line for line in result_file.readlines() if "TESTRESULT" in line]
+            result_list = [line.split(':')[1].strip()
+                           for line in result_file.readlines() if "result:" in line]
         if not result_list:
             raise tmt.utils.ExecuteError(
                 f"Test result not found in result file '{report_result_path}'.")
-        result = result_list[0].split("=")[1].strip()
+        num_result = [hierarchy[line] for line in result_list]
+        result = list(hierarchy.keys())[list(hierarchy.values()).index(max(num_result))]
```

Comment:
The `list` conversion of keys and values looks suspicious. It wasn't clear to me what the code does, so I spent some time thinking of how to simplify it, and, as it relies on the order of keys and values of a dictionary - which is now part of the Python standard, yet a bit unusual - I was wondering whether a simple list would be enough :

```python
hierarchy = ['skip', 'pass', 'warn', 'fail']

with open(report_result_path) as result_file:
    result_indices = [
        hierarchy.index(line.split(':')[1].strip())
        for line in result_file.readlines() if "result:" in line
    ]

if not result_indices:
    raise tmt.utils.ExecuteError(f"Test result not found in result file '{report_result_path}'.")

# The final result is that of collected results with the highest index in our hierarchy
result = hierarchy[hierarchy.index(max(result_indices))]
```

Reply:
@pfdaly PTAL pls

Reply:
I believe this one is covered. @happz is that right?